DEVELOPMENT.md

```diff
@@ -40,33 +40,47 @@ Automated `unit` tests can be written using [vscode-tmgrammar-test](https://gith
 To start the test suite from the command-line run:
 
 ```bash
-npm test:unit
+npm test
 ```
 
-## Writing Grammar Unit Tests
-
-Unit tests are Terraform files with `vscode-tmgrammar-test` token lines specifying which TextMate grammer should be resolved.
-
-For example:
+## Writing Grammar Snapshot Tests
 
-> Note: This is shortened to demonstrate, actual syntax will vary
+Snapshot tests comprise of two files: example files and their companion `snap` file:
 
+```bash
+tests/snapshot/terraform/blocks.tf
+tests/snapshot/terraform/blocks.tf.snap
 ```
-resource "example" "thing" {
-;         ^^^^^^^ source.terraform string.quoted.double.terraform
-;                   ^^^^^ source.terraform string.quoted.double.terraform
-;                          ^ source.terraform punctuation.section.block.begin.terraform
-}
-; <- source.terraform punctuation.section.block.end.terraform
 
-```
+These files are grouped by HashiCorp product inside folders named for the product they target.
 
-## Writing Grammar Snapshot Tests
+Snapshot test example files are HashiCorp product files without any `vscode-tmgrammar-test` token lines. Each example file is exactly how you would see it used in production. This ensures scope, inheritance, and resolution of tokens happen exactly as they would on a user's machine.
 
-Snapshot tests comprise of two files: example files and their companion `snap` file.
+The companion `snap` file is named the same as the example file with the `.snap` extension, and is the tmGrammar represenation of all resolved tokens. This file is commited alongside the example file. If anything changes with regards to how the tokens are resolved, the snapshot test will fail.
```

Comment:
```suggestion
The companion `snap` file is named the same as the example file with the `.snap` extension, and is the tmGrammar representation of all resolved tokens. This file is committed alongside the example file. If anything changes with regards to how the tokens are resolved for these example files, the snapshot test will fail.
```

Reply:
This would otherwise sound like the tests are more fragile than they are. 😃 I mean - they _are_ fragile, but if the case isn't covered by tests, then the tests should continue to pass, right?

Reply:
Snapshot tests fail if the generated output is different, so they are fragile on purpose. For example, we expect a given line to have 5 tokens and with a change it now has 7 tokens, the test fails because we did not cover it. Simiarily with less tokens or different tokens etc.